micro-rdk/src/common/ota.rs

```diff
@@ -220,12 +220,34 @@ impl<S: OtaMetadataStorage> OtaService<S> {
 
         let connector = OtaConnector::default();
 
+        #[cfg(not(feature = "esp32"))]
+        let (max_size, address) = (1024 * 1024 * 4, 0xabcd);
+        #[cfg(feature = "esp32")]
+        let (max_size, address) = {
+            log::debug!("getting handle to next OTA update partition");
+            let ptr: *const esp_partition_t =
+                unsafe { esp_ota_get_next_update_partition(std::ptr::null()) };
+
+            if ptr.is_null() {
+                log::warn!("pointer to next update partition was null");
```

Comment:
- Maybe add "device may not be correctly partitioned for OTA" to the message.
- Similar thoughts on message in the `OtaError` below.
- Maybe use the same message for both? Do you have a way to log the message from an error?
```
let err = OtaError::Other("some good message here);
log::warn!(err.to_string())
return err
```

Reply:
added it. though we already log the returned error [here](https://github.com/viamrobotics/micro-rdk/blob/a5d7a4a0a21a51c932b61e590796ab6f6193b229/micro-rdk/src/common/conn/viam.rs#L554), so you might see it twice now.